src/routes/embed/projects/[project_id]-[slug]/+page.ts

```diff
@@ -11,28 +11,31 @@ const OLD_PATTERN = /(\D+)-(\d+)/;
 
 /** @type {import('./$types').PageLoad} */
 export async function load({ params, fetch, url, setHeaders }) {
-  // handle old URLs
   let { project_id, slug }: Record<string, Maybe<string>> = params;
-  if (`${project_id}-${slug}`.match(OLD_PATTERN)) {
-    const groups = OLD_PATTERN.exec(`${project_id}-${slug}`);
-    project_id = groups?.[2];
-  }
-
-  if (!project_id) {
-    return error(404, "Project not found");
-  }
-
-  const project = await get(+project_id, fetch);
-  const documents = search(`+project:${project_id}`, undefined, fetch);
+  let project = await get(+project_id, fetch);
 
+  // old project embeds used a pattern of <slug>-<id>
+  // so we want to give 404s a chance to try again
   if (project.error) {
```

Comment:
Can we branch on `project.error.status === 404` so we're not making extra requests for 403, 500, etc?

Reply:
Good call. Done.